heisskleber/core/async_factories.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+from heisskleber.config import BaseConf, load_config
+from heisskleber.mqtt import AsyncMqttPublisher, AsyncMqttSubscriber, MqttConf
+from heisskleber.zmq import ZmqAsyncPublisher, ZmqAsyncSubscriber, ZmqConf
+
+from .types import AsyncSink, AsyncSource
+
+_registered_async_sinks: dict[str, tuple[type[AsyncSink], type[BaseConf]]] = {
+    "mqtt": (AsyncMqttPublisher, MqttConf),
+    "zmq": (ZmqAsyncPublisher, ZmqConf),
+}
+
+_registered_async_sources: dict[str, tuple] = {
+    "mqtt": (AsyncMqttSubscriber, MqttConf),
+    "zmq": (ZmqAsyncSubscriber, ZmqConf),
+}
+
+
+def get_async_sink(name: str) -> AsyncSink:
+    """
+    Factory function to create a sink object.
+
+    Parameters:
+        name: Name of the sink to create.
+        config: Configuration object to use for the sink.
+    """
+
+    if name not in _registered_async_sinks:
+        error_message = f"{name} is not a registered Sink."
+        raise KeyError(error_message)
+
+    pub_cls, conf_cls = _registered_async_sinks[name]
+
+    if "MSB_CONFIG_DIR" in os.environ:
```

Comment:
from my point of view, we can remove any reference to any environment vars as we are defaulting to `$HOME/.config/heisskleber`